app/Controllers/previewController.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * Controller to handle every preview content.
+ */
+class FreshRSS_preview_Controller extends Minz_ContentController {
+
+	public function init() {
+		if (!FreshRSS_Auth::hasAccess()) {
+			Minz_Error::error(403);
+		}
+	}
+
+
+	/**
+	 * This content creates a preview of a content-path.
+	 *
+	 * Parameters are:
+	 *   - id (mandatory - no default): Feed ID
+	 *   - path (mandatory - no default): Path to preview
+	 *
+	 */
+	public function pathQueryContent() {
+		header('Content-Type: text/html; charset=UTF-8');
+
+		//Get parameters.
+		$feed_id = Minz_Request::param('id');
+		$content_path = Minz_Request::param('path');
+
+		if (!$content_path) {
+			echo 'path-not-set'; // FIXME: translate.
+			return;
+		}
+
+		//Check Feed ID validity.
+		$entryDAO = FreshRSS_Factory::createEntryDao();
+		$entries = $entryDAO->listWhere('f', $feed_id);
+
+		if (count($entries) == 0) {
+			echo 'no-entries-found'; // FIXME: translate.
+			return;
+		}
+
+		$entry = $entries[0];
+		$feed = $entry->feed(true);
+
+		if (!$feed) {
+			echo 'no-feed-for-entry'; // FIXME: translate.
+			return;
+		}
+
+		$feed->_pathEntries($content_path);
+		$entry->loadCompleteContent(true);
+
+		echo '<!DOCTYPE html><html><body>';
+		echo $entry->content();
+		echo '</body></html>';
```

Comment:
Is there a reason why you echo the content instead of using a template?

Reply:
Not completely sure of your point but the content, there, is going to be (most of the time) raw partial HTML structure extracted from the website itself (and not the feed content). So it doesn't fit well with our templating system (?).
And anyway, the style while depends on the RSS client (FreshRRS Web UI, Reeder, etc.) showing it, so I'm not sure it makes sense to force a style ?

Or, if you mean that the enclosure (`<html><body>…`) should be in a template, then why not. No special reason, except I wasn't aware of this mechanism.

Or, if you mean that using the `echo` function is not a good idea for some reason, then same: no special reason, except that I wasn't aware of a better mechanism.

Reply:
The source may be more useful than the rendered result (not saying you can't have both), as in something like `htmlspecialchars($entry->content())`.

Reply:
@Frenzie I'm not sure to understand :s

Anyway, I guess that suggestion from @marienfressinaud will fix this concern ?

Reply:
If I understand, the suggestion of @Frenzie is to show the raw HTML instead of the rendered, but I don't agree. We might have a toggle to switch (a bit like the GitHub comment editor), but as a user I largely prefer to "see" what I'll get instead of some HTML

Reply:
Ah yes, I see. So well, having the raw HTML can have interest to help write the path, indeed. But yes, having the actual rendered result look better for an end-user.

I will check if I can add the two (like the rendered first, and the raw after), or even having some web UI to switch between the two.

Reply:
@marienfressinaud I explicitly didn't say instead. ;-)

Reply:
Okay, I did something which works. Not sure it's the best way to do it, but look somehow ok to me, for a preview at least.